stages/org.osbuild.rhsm.facts

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+"""
+The `org.osbuild.rhsm.facts` stage is allows for storing rhsm-facts format
+artifacts in the built image. Keys and values are supplied in options.
+"""
+
+import json
+import os
+import sys
+
+import osbuild.api
+
+
+SCHEMA = r"""
+"definitions": {
+  "key": {
+    "type": "string"
+  },
+  "value": {
+    "type": "string"
+  }
+},
+"properties": {
+  "facts": {
+    "type": "array",
+    "description": "Array of key/value facts.",
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "key": {
+          "$ref": "#/definitions/key"
+        },
+        "value": {
+          "$ref": "#/definitions/value"
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    path = os.path.join(tree, "usr", "share", "osbuild", "self")
+    file = os.path.join(path, "rhsm.facts")
+
+    os.makedirs(path, exist_ok=True)
+
+    with open(file, "w", encoding="utf-8") as f:
+        json.dump({fact["key"]: fact["value"] for fact in options["facts"]}, f)
+
+    os.makedirs(os.path.join(tree, "etc", "rhsm", "facts"), exist_ok=True)
+
+    os.symlink(
+        os.path.join("/usr/share/osbuild/self/rhsm.facts"),
+        os.path.join(tree, "etc", "rhsm", "facts", "osbuild.facts"),
```

Comment:
You use `join` quite inconsistently, imo. Either be fine with manual concatenation, or join all of them. My preferred style is `os.path.join(tree, "etc/rhsm/facts/osbuild.facts")`, but regardless what you decide, the first argument here should be consistent with the second, shouldn't it?

Also, if you define `path = "usr/share/osbuild/self"`, you can use it here and in `os.makedirs()`.